Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp

```diff
@@ -484,6 +484,11 @@ void Device::UpdateInput()
   {
     if (auto pad_data = msg.CheckAndCastTo<Proto::MessageType::PadDataResponse>())
     {
+      if (pad_data->pad_id != m_index)
+      {
+        continue;
```

Comment:
While this probably fixes the issue, having each device instance read from the one socket and dropping irrelevant messages seems very wrong. You're going to have a much less responsive controller if each device is reading and dropping the other devices' data. There probably needs to be a single reader of the socket or messages need to be peeked at before they are processed.

Reply:
Hmm, never thought of it that way.  Having a server specific reading socket makes a lot of sense.  I'll remove this and we can fix it proper later on.  I'm thinking you could have a socket per server that fans out to the devices.

Reply:
> ... having each device instance read from the one socket and dropping irrelevant messages seems very wrong. You're going to have a much less responsive controller if each device is reading and dropping the other devices' data.

Each device has its own socket connection to the server.  My UDP knowledge is extremely shaky, but I think each socket is bound to a separate local port. So I don't think reading one socket drops data on another socket. See my [comment](https://bugs.dolphin-emu.org/issues/12052#note-6) on the bug tracker for related info.

Reply:
@rlnilsen Ah. If each device has its own connection then it should work. But this seems like a sloppy way of doing things, duplicating the bandwidth for each device for no reason? Or is the protocol smart enough to send data for specific devices?

Reply:
> Or is the protocol smart enough to send data for specific devices?

Exactly. Except the DSU Server used in the [bug report](https://bugs.dolphin-emu.org/issues/12052) that prompted this code change was not. It sent data from all connected controllers even if data from only one was requested.